
#!/usr/bin/env python


PACKAGE = "scout_odom"
NODE = "odomEstimator"

from dynamic_reconfigure.parameter_generator_catkin import *
import math


gen = ParameterGenerator()

gen.add("x_init", double_t, 0,"Initial position on x", default=0.0)
gen.add("y_init", double_t, 1,"initial position on y", default=0.0)
gen.add("theta_init", double_t, 2,"Initial orientation on z", default=0.0)

int_enum = gen.enum([ gen.const("Euler", int_t, 0, "Euler integration method"),
					  gen.const("Runge_Kutta", int_t, 1, "RungeKutta integration method")], 
					  "Integration methods")


gen.add("int_method", int_t, 3, "Integration methods supported for odometry computation", 0, 0, 1, edit_method=int_enum)

exit(gen.generate(PACKAGE, NODE, "odomParams"))

